mlflow/shap.py

```diff
@@ -179,3 +257,334 @@ def predict_function(X) -> pred:
+
+        if underlying_model_flavor == "sklearn":
+            mlflow.sklearn.save_model(explainer.model.model.__self__, underlying_model_path)
+        elif underlying_model_flavor == "pytorch":
+            mlflow.pytorch.save_model(explainer.model.model, underlying_model_path)
+
+    # saving the explainer object
+    explainer_data_subpath = "explainer.shap"
+    explainer_output_path = os.path.join(path, explainer_data_subpath)
+    with open(explainer_output_path, "wb") as explainer_output_file_handle:
+        explainer.save(explainer_output_file_handle)
+
+    conda_env_subpath = "conda.yaml"
+    if conda_env is None:
+        conda_env = get_default_conda_env()
+    elif not isinstance(conda_env, dict):
+        with open(conda_env, "r") as f:
+            conda_env = yaml.safe_load(f)
+
+    # merging the conda environment generated by serializing the underlying model
+    if underlying_model_path is not None:
+        underlying_model_conda_path = os.path.join(underlying_model_path, "conda.yaml")
+        with open(underlying_model_conda_path, "r") as underlying_model_conda_file:
+            underlying_model_conda_env = yaml.safe_load(underlying_model_conda_file)
+        conda_env = _merge_environments(conda_env, underlying_model_conda_env)
+
+    with open(os.path.join(path, conda_env_subpath), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    pyfunc.add_to_model(
+        mlflow_model,
+        loader_module="mlflow.shap",
+        model_path=explainer_data_subpath,
+        underlying_model_flavor=underlying_model_flavor,
+        env=conda_env_subpath,
+    )
+
+    mlflow_model.add_flavor(
+        FLAVOR_NAME,
+        shap_version=shap.__version__,
+        serialized_explainer=explainer_data_subpath,
+        underlying_model_flavor=underlying_model_flavor,
+    )
+
+    mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
+
+
+def _get_conda_and_pip_dependencies(conda_env):
+    """
+    Extract conda and pip dependencies from conda environments
+
+    :param conda_env: Conda environment
+    """
+
+    conda_deps = []
+    pip_deps = []
+
+    for dependency in conda_env["dependencies"]:
+        if isinstance(dependency, dict) and dependency["pip"]:
+            for pip_dependency in dependency["pip"]:
+                if pip_dependency != "mlflow":
+                    pip_deps.append(pip_dependency)
+        else:
+            if dependency.split("=")[0] != "python" and dependency.split("=")[0] != "pip":
+                conda_deps.append(dependency)
+
+    return conda_deps, pip_deps
+
+
+def _merge_environments(shap_environment, model_environment):
+    """
+    Merge conda environments of underlying model and shap.
+
+    :param shap_environment: SHAP conda environment.
+    :param model_environment: Underlying model conda environment.
+    """
+
+    merged_conda_channels = list(set(shap_environment["channels"] + model_environment["channels"]))
```

Comment:
```suggestion
    merged_conda_channels = list(set(shap_environment["channels"] + model_environment["channels"]) - set(["default", "conda-forge"]))
```

Can we remove `default` and `conda-forge`?